src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -396,6 +397,50 @@ private void VerifyIsAlive()
                 throw new InvalidOperationException(SR.AssemblyLoadContext_Verify_NotUnloading);
             }
         }
+
+        private static readonly AsyncLocal<AssemblyLoadContext> _asyncLocalCurrentContextualReflectionContext = new AsyncLocal<AssemblyLoadContext>();
+
+        public static AssemblyLoadContext CurrentContextualReflectionContext
+        {
+            get { return _asyncLocalCurrentContextualReflectionContext.Value; }
+        }
+
+        public ContextualReflectionScope EnterContextualReflection()
+        {
+            return new ContextualReflectionScope(this);
+        }
+
+        static public ContextualReflectionScope EnterContextualReflection(Assembly activating)
```

Comment:
```suggestion
        public static ContextualReflectionScope EnterContextualReflection(Assembly activating)
```

The convention is to have access modifier first.